src/Classes/ServiceAPI/MyRadio_User.php

```diff
@@ -629,6 +636,20 @@ public function getAuthProvider()
     }
 
     /**
+     * Get whether the user has signed the Presenter Contract
+     *
+     * @return bool if the presenter has signed the contract
+     */
+    public function hasSignedContract()
+    {
+        if (empty(Config::$contract_uri)) {
+            return true;
+        } else {
+            return $this->contract_signed;
```

Comment:
I don't see any change to the constructor to pull this value from the database, so it's undefined?